packages/oidc-client/src/lib/client.store.ts

```diff
@@ -301,6 +301,83 @@ export async function oidc<ActionType extends ActionTypes = ActionTypes>({
           };
         }
       },
+      /**
+       * @method revoke
+       * @description Revokes an access token using the revocation endpoint from the wellknown configuration.
+       *              It requires an access token stored in the configured storage.
+       * @returns {Promise<GenericError | RevokeResult>} - Returns a promise that resolves to the revoke response or an error response.
+       */
+      revoke: async (): Promise<GenericError | RevokeResult> => {
+        const state = store.getState();
+        const wellknown = wellknownSelector(wellknownUrl, state);
+
+        if (!wellknown?.revocation_endpoint) {
+          return {
+            error: 'Wellknown missing revocation endpoint',
+            type: 'wellknown_error',
+          };
+        }
+
+        const tokens = await storageClient.get();
+
+        if (!tokens || !('accessToken' in tokens)) {
+          return {
+            error: 'No access token found',
+            type: 'state_error',
+          };
+        }
+
+        const revokeµ = Micro.promise(() =>
+          store.dispatch(
+            oidcApi.endpoints.revoke.initiate({
+              accessToken: tokens.accessToken,
+              clientId: config.clientId,
+              endpoint: wellknown.revocation_endpoint,
+            }),
+          ),
+        ).pipe(
+          Micro.map(({ error }) => {
+            if (error) {
+              let message = 'An error occurred while revoking the token';
+              let status: number | string = 'unknown';
+              if ('message' in error && error.message) {
+                message = error.message;
+              }
+              if ('status' in error) {
+                status = error.status;
+              }
+              return {
+                error: 'Token revocation failure',
+                message,
+                type: 'auth_error',
+                status,
+              } as GenericError;
+            }
+
+            return null;
+          }),
+          // Delete local token and return combined results
+          Micro.flatMap((revokeResponse) =>
+            Micro.promise(() => storageClient.remove()).pipe(
```

Comment:
This is good.

Semantically, it may "feel" cleaner if we wrap both effects in a pipe, but not a big deal.

Alternative being:

```ts
pipe(
    Micro.promise(() => storageClient.remove()),
    Micro.map((deleteResponse) => ({ revokeResponse, deleteResponse }))
 )
```   

This is entirely subjective so i'll leave it to you what you feel reads better they are functionally equivalent if i have my mental model correct.